### DISCUSSION TITLE
RTC time lost on reset.

### DISCUSSION THREAD

**Discussion by ADenislebel** [micropython/micropython#11217] [RP2040 / Pico]
This is as much a comment as a suggestion to the powers that be about the RTC clock at least with the RP2 board. I found out the hard (no pun intended) way that upon a hard reset the RTC time is lost. I wish it were otherwise but since I believe the stock Pico does not have any EEprom, my question: Is there an easy solution the storing the RTC tuple in flash or otherwise where so it could be retrieved after a reset. 

Regards Denis Lebel

**Comment by robert-hh** [comment]
Hard(ware) reset is completely independent from software.  To cope with that, one would have to store the time at regular frequent intervals in flash. which would wear out flash pretty fast. And still there would be time lags. That's why external RTC devices usually have a small battery to store the actual time value in RAM and to keep the RTC running. Some MCU like STM32 or MIMXRT support that as well with their internal RTC. I do not know if the RP2040 can do that.

**Reply by scruss** [reply]
One of the flaws in the RP2040's design is that its RTC has no ability to take battery backup. The time is always lost on reset. 

If you need persistent time, you need to add an external RTC + battery module like a DS3231 or DS1307

**Comment by tabemann** [comment]
The RP2040 does not support retaining its RTC clock date/time across resets, unfortunately. I too wish that its clock would stay set across resets as long as it were powered.

**Comment by peterhinch** [comment]
DS3231 boards with backup coin cell are readily available on eBay. [This doc](https://github.com/peterhinch/micropython-samples/tree/master/DS3231) provides code and details on using them.

**Comment by ADenislebel** [comment]
Thanks guys I'm a hardware  guy and of course I could use an external RTC.
For now I added a battery backup to the Pico but without
a charging circuit its going to die pretty quick if power goes down. I
agree the RTC without retention on bootup is kind of sad.
Regards Denis Lebel
